compiler_gym/datasets/datasets.py

```diff
@@ -0,0 +1,480 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+import warnings
+from typing import Dict, Iterable, Optional, Set, Union
+
+import numpy as np
+
+from compiler_gym.datasets.benchmark import (
+    BENCHMARK_URI_RE,
+    Benchmark,
+    resolve_uri_protocol,
+)
+from compiler_gym.datasets.dataset import Dataset
+
+
+class Datasets(object):
+    """A collection of datasets.
+
+    The Datasets class serves two purposes:
+
+    1. to provide a convenient interface for selecting benchmarks from multiple
+       datasets; and
+
+    2. to make it easier to manage multiple datasets by decoupling the selection
+       of datasets from their physical installation.
+
+    For the first goal, methods such as :meth:`benchmark()` provide a convenient
+    interface for randomly selecting benchmarks across multiple datasets:
+
+        >>> for i in range(100):
+        ...     benchmark = env.datasets.benchmark()
+
+    To iterate over a specific dataset, the datasets object can be indexed:
+
+        >>> for i in range(100):
+        ...     benchmark = env.datasets["cbench-v1"].benchmark()
+
+    The datasets object can be also be iterated over to list the available
+    datasets:
+
+        >>> for dataset in env.datasets:
+        ...     print(dataset.name)
+        benchmark://cbench-v1
+        benchmark://github-v0
+        benchmark://npb-v0
+
+    For the second goal, the datasets interface adds an *active* vs *installed*
```

Comment:
IMO this is a weird thing to have. I'm not convinced I understand the use case.
Why not have everything be installed and then layer something over the top as a separate class - for the few use cases that might need this.
E.g. if I want to do something to half the benchmarks I have access to write a filter.

I'd suggest dropping the second goal.